app.rb

```diff
@@ -47,7 +47,7 @@ def openid_configuration
 
   def token(code)
     json HTTParty.post(
-      openid_configuration[:token_endpoint],
+      ENV['IDP_TOKEN_URL'] || openid_configuration[:token_endpoint],
```

Comment:
I would do something like this: https://github.com/18F/identity-openidconnect-sinatra/compare/margolis-basic-auth-config

The only thing missing is a spec to make sure it doesn't set basic auth creds when the env is not there I think